tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,134 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integrationoperator
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Operator", Ordered, func() {
+	SetDefaultEventuallyTimeout(60 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+
+	Describe("operator installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("default helm manifest are applied", func() {
+			BeforeEach(func() {
+				if skipDeploy == "true" {
+					Skip("Skipping the deployment of the operator and the tests")
+				}
+
+				Eventually(deployOperator).Should(Succeed())
+
+			})
+
+			Specify("the operator is running", func() {
+				Eventually(getConditionStatus).WithArguments(namespace, "deployment", deploymentName, "Available").Should(Equal("True"))
+				podName, _ := getPodNameFromLabel(namespace, "control-plane=istio-operator")
+				Eventually(getPodPhase).WithArguments(namespace, podName).Should(Equal("Running"))
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		for _, version := range istioVersions {
+			// This var version is needed to avoid the closure of the loop
+			version := version
+
+			Context(fmt.Sprintf("is applied the istio resource with version %s", version), func() {
+				BeforeAll(func() {
+					Expect(createNamespaceIfNotExists(controlPlaneNamespace)).To(Succeed())
+					Eventually(createIstioCR).WithArguments(version).Should(Succeed())
+				})
+
+				AfterAll(func() {
+					Eventually(deleteNamespace).WithArguments(controlPlaneNamespace).Should(Succeed())
+					Eventually(namespaceIsDeleted).WithArguments(controlPlaneNamespace).Should(Equal(true))
+				})
+
+				When("the Istio resource is created", func() {
+					It("updates the Istio resource status to Ready and Running", func() {
+						Eventually(getConditionStatus).WithArguments(controlPlaneNamespace, "istio", istioName, "Reconciled").Should(Equal("True"))
+						Eventually(getConditionStatus).WithArguments(controlPlaneNamespace, "istio", istioName, "Ready").Should(Equal("True"))
+
+						istiodPodName, _ := getPodNameFromLabel(controlPlaneNamespace, "app=istiod")
+						Eventually(getPodPhase).WithArguments(controlPlaneNamespace, istiodPodName).Should(Equal("Running"))
+					})
+
+					Specify("the istio resource version match the applied version", func() {
+						Expect(getInstalledIstioVersion()).Should(ContainSubstring(version))
+					})
+
+					It("istio resource stopped reconciling", func() {
+						Eventually(captureIstiodLast30SecondsOfLog).WithTimeout(120 * time.Second).Should(ContainSubstring("Reconciliation done"))
```

Comment:
This doesn't check what it used to check. The last 30s of log should *not* contain "Reconciliation done".

Reply:
Ohh, I see it. Fixed